app/finders/notification_finder.rb

```diff
@@ -26,20 +26,41 @@ def count
 
   def set_up
     find_all_notifications
-    filter_by_read_status
-    filter_by_status
+    apply_filters
   end
 
   def find_all_notifications
     @notifications = current_user.notifications.where(account_id: @current_account.id)
   end
 
-  def filter_by_status
-    @notifications = @notifications.where('snoozed_until > ?', DateTime.now.utc) if params[:status] == 'snoozed'
+  def apply_filters
+    status = params[:status]
+    type = params[:type]
+    # Return all the notifications including read, unread, and snoozed.
+    return if status == 'snoozed' && type == 'read'
+
+    # Return only read and unread notifications, and do not display snoozed notifications
+    if type == 'read'
+      exclude_snoozed_notifications
+    # Return only snoozed and unread notifications, and do not display read notifications
+    elsif status == 'snoozed'
+      exclude_read_notifications
+    else
+      # Default case: return all the unread notifications
+      include_unread_notifications
+    end
+  end
+
+  def exclude_snoozed_notifications
+    @notifications = @notifications.where(snoozed_until: nil)
+  end
+
+  def exclude_read_notifications
+    @notifications = @notifications.where(read_at: nil)
   end
 
-  def filter_by_read_status
-    @notifications = @notifications.where.not(read_at: nil) if params[:type] == 'read'
+  def include_unread_notifications
+    @notifications = @notifications.where('snoozed_until IS NULL AND read_at IS NULL')
   end
```

Comment:
Can't we simplify this to the following? 

```rb

def set_up
  find_all_notifications
  filter_snoozed_notifications
  fitler_read_notifications
end

def find_all_notifications
  @notifications = current_user.notifications.where(account_id: @current_account.id)  
end

def filter_snoozed_notifications
  @notifications = @notifications.where.not(snoozed_until: nil) if params[:status].empty?
end

def fitler_read_notifications
  @notifications = @notifications.where.not(read_at: nil) if params[:type] != 'read'
end
```

Reply:
I would also suggest to change the params to something like the following.

Instead of 
```
/notifications?type=read&status=snoozed
```

let us change it to

```
/notifications?includes[]=snoozed&includes[]=read
```

This would make the filters easy to understand.

The finder would be simplified to

```
class NotificationFinder
  def set_up
    find_all_notifications
    filter_snoozed_notifications
    fitler_read_notifications
  end

  def find_all_notifications
    @notifications = current_user.notifications.where(account_id: @current_account.id)  
  end

  def filter_snoozed_notifications
    @notifications = @notifications.where(snoozed_until: nil) unless type_included?('snoozed')
  end

  def fitler_read_notifications
    @notifications = @notifications.where(read_at: nil) unless type_included?('read')
  end

  def type_included?(type)
    (params[:includes] || []).includes?(type)
  end
end
```